### PR TITLE
Set ELBConnectionTimeout on IngressController for v4.11+

### PR DESCRIPTION
**Pull request by blrm**
As of OCP v4.11, cluster-ingress-operator allows management of AWS ELB idle connection timeouts by setting them in the spec of the IngressController: https://github.com/openshift/cluster-ingress-operator/pull/451 . Since cloud-ingress-operator currently sets that idle timeout by annotating the LoadBalancer service object directly, it was causing a fight between the two operators were cloud-ingress-operator was applying the annotation to the service and cluster-ingress-operator was removing it, causing frequent AWS calls as the setting was modified back and forth.

This PR updates the publishingstrategy controller to set the ELB idle timeout directly in the IngressController for v4.11+ clusters, and to patch any IngressControllers owned by cloud-ingress-operator that lack that field in their spec. 

I tested this against 4.10 and 4.11 clusters. On 4.10, the current behavior still worked correctly and the cloud-ingress-operator makes no attempt to set the connectionIdleTimeout on the IngressController. On 4.11, it correctly patched the default IngressController to set the idle connection timeout there, and once set it no longer competes with the cluster-ingress-operator to set the annotation on the LB service object.

### CODE REVIEW

**Review by fahlmant — APPROVED** [review]
/lgtm
/approve

I'm going to add
/hold
as I'd like at least one person from Aurora to also sign off :)

**Comment by lnguyen1401** [comment]
Thank you for fixing this !

**Comment by blrm** [comment]
Thanks @lnguyen1401 and @fahlmant! Now that Linh has lgtm'ed as well, I'm going to drop the hold.

/unhold

**Pull request merged**
